types/index.d.ts

```diff
@@ -0,0 +1,369 @@
+    ? Factories[K] extends FactoryDefinition<infer Data>
+      ? FlattenFactoryMethods<Data>
+      : {}
+    : {};
+
+  /**
+   * Models all available information about a given set of model and
+   * factory definitions, determining the behavior of ORM methods on
+   * a `Server` and its corresponding `Schema` instance.
+   */
+  export type Registry<
+    Models extends Record<string, ModelDefinition>,
+    Factories extends Record<string, FactoryDefinition>
+  > = {
+    [K in keyof Models | keyof Factories]: ModelInstance<
+      ExtractModelData<Models, K> & ExtractFactoryData<Factories, K>
+    >;
+  };
+
+  /** Represents the type of an instantiated Mirage model.  */
+  export type ModelInstance<Data = {}> = Data & {
+    id?: string;
+    attrs: Record<string, unknown>;
+    modelName: string;
+
+    /** Persists any updates on this model back to the Mirage database. */
+    save(): void;
+
+    /** Updates and immediately persists a single attr on this model. */
+    update<K extends keyof Data>(key: K, value: Data[K]): void;
+
+    /** Removes this model from the Mirage database. */
+    destroy(): void;
+
+    /** Reloads this model's data from the Mirage database. */
+    reload(): void;
+  };
+}
+
+declare module "miragejs/server" {
+  import { Request, Response } from "miragejs";
+  import { ModelInstance } from "miragejs/-types";
+  import Db from "miragejs/db";
+  import Schema from "miragejs/orm/schema";
+
+  type MaybePromise<T> = T | PromiseLike<T>;
+
+  /** A callback that will be invoked when a given Mirage route is hit. */
+  export type RouteHandler<Registry> = (
+    schema: Schema<Registry>,
+    request: Request
+  ) => MaybePromise<ModelInstance | Response | object>;
+
+  export interface HandlerOptions {
+    /** A number of ms to artifically delay responses to this route. */
+    timing?: number;
+  }
+
+  export class Server<Registry = Record<string, ModelInstance>> {
+    /** The underlying in-memory database instance for this server. */
+    readonly db: Db;
+
+    /** An interface to the Mirage ORM that allows for querying and creating records. */
+    readonly schema: Schema<Registry>;
+
+    /** Creates a model of the given type. */
+    readonly create: Schema<Registry>["create"];
+
+    /** Whether or not Mirage should log all requests/response cycles. */
+    logging: boolean;
+
+    /** A default number of ms to artifically delay responses for all routes. */
+    timing: number;
+
+    /** A default prefix applied to all subsequent route definitions. */
+    namespace: string;
+
+    /** Creates multiple models of the given type. */
+    createList<
+      K extends keyof Registry,
+      Init extends Registry[K],
+      Data extends Partial<Init>
+    >(modelName: K, count: number, data?: Data): Array<Init & Data>;
+
+    /** Handle a GET request to the given path. */
+    get(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
```

Comment:
`Server` also has the `options` method (for the `OPTIONS` request in HTTP)

Reply:
Thanks, added.